karapace/schema_backup.py

```diff
@@ -134,23 +134,31 @@ def request_backup(self):
         if not self.consumer:
             self.init_consumer()
         self.log.info("Starting schema backup read for topic: %r", self.topic_name)
+
         values = []
-        raw_msg = self.consumer.poll(timeout_ms=self.timeout_ms)
-        for _, messages in raw_msg.items():
-            for message in messages:
-                key = message.key.decode("utf8")
-                try:
-                    key = json.loads(key)
-                except json.JSONDecodeError:
-                    self.log.debug("Invalid JSON in message.key: %r, value: %r", message.key, message.value)
-                value = None
-                if message.value:
-                    value = message.value.decode("utf8")
+        topic_fully_consumed = False
+
+        while not topic_fully_consumed:
+
+            raw_msg = self.consumer.poll(timeout_ms=self.timeout_ms)
+            topic_fully_consumed = len(raw_msg) == 0
+
+            for _, messages in raw_msg.items():
+                for message in messages:
+                    key = message.key.decode("utf8")
                     try:
-                        value = json.loads(value)
+                        key = json.loads(key)
                     except json.JSONDecodeError:
-                        self.log.debug("Invalid JSON in message.value: %r, key: %r", message.value, message.key)
-                values.append((key, value))
+                        self.log.debug("Invalid JSON in message.key: %r, value: %r", message.key, message.value)
+                    value = None
+                    if message.value:
+                        value = message.value.decode("utf8")
+                        try:
+                            value = json.loads(value)
+                        except json.JSONDecodeError:
+                            self.log.debug("Invalid JSON in message.value: %r, key: %r", message.value, message.key)
+                    values.append((key, value))
```

Comment:
Can we end up adding an invalid json byte blob as a value?

Reply:
It looks so, for both keys and values. What are the expectations here? Should that ever happen? If it does, how should it be handled?

Reply:
For now I think we can leave it as it is for debugging purposes. The restore part of this tool needs some love before it's going to be used for real.